File: src/jaslib_src/jaslib/ui/web/display_type.py
from __future__ import annotations


class DisplayType:
    """Display type constants and checks. Portable to C#."""
    REVIEW_QUESTION: str = "reviewQuestion"
    REVIEW_ANSWER: str = "reviewAnswer"
    _ANSWER_TYPES: set[str] = {"reviewAnswer", "previewAnswer", "clayoutAnswer"}

    @staticmethod
    def is_displaying_answer(display_type: str) -> bool:
        return display_type in DisplayType._ANSWER_TYPES

    @staticmethod
    def is_displaying_review_question(display_type: str) -> bool:
        return display_type == DisplayType.REVIEW_QUESTION

    @staticmethod
    def is_displaying_review_answer(display_type: str) -> bool:
        return display_type == DisplayType.REVIEW_ANSWER

File: src/src_dotnet/JAStudio.Core/UI/Web/DisplayType.cs
using System.Collections.Generic;

namespace JAStudio.Core.UI.Web;

/// <summary>
/// Display type constants and checks.
/// </summary>
static class DisplayType
{
   public const string ReviewQuestion = "reviewQuestion";
   public const string ReviewAnswer = "reviewAnswer";

   static readonly HashSet<string> QuestionTypes = ["reviewQuestion", "previewQuestion", "clayoutQuestion"];
   static readonly HashSet<string> AnswerTypes = ["reviewAnswer", "previewAnswer", "clayoutAnswer"];

   public static bool IsDisplayingQuestion(string displayType) => QuestionTypes.Contains(displayType);

   public static bool IsDisplayingAnswer(string displayType) => AnswerTypes.Contains(displayType);

   public static bool IsDisplayingReviewQuestion(string displayType) => displayType == ReviewQuestion;

   public static bool IsDisplayingReviewAnswer(string displayType) => displayType == ReviewAnswer;

   public static bool IsReview(string displayType) => displayType.StartsWith("review");
}
